src/renderer/hooks/useMapStylesQuery.js

```diff
@@ -0,0 +1,69 @@
+// @ts-check
+import { useExperimentsFlagsStore } from './store'
+import { useMapServerQuery } from './useMapServerQuery'
+import { useQuery } from '@tanstack/react-query'
+import api from '../new-api'
+import { defineMessages, useIntl } from 'react-intl'
+
+// Randomly generated, but should not change, since this is stored in settings
+// if the user selects one of these "legacy" map styles
+const DEFAULT_MAP_ID = '487x2pc8ws801avhs5hw58qnxc'
+const CUSTOM_MAP_ID = 'vg4ft8yvzwfedzgz1dz7ntneb8'
+const ONLINE_STYLE_URL = 'mapbox://styles/mapbox/outdoors-v10'
+
+const m = defineMessages({
+  // The name of the default background map
+  defaultBackgroundMapName: 'Default',
+  // The name of the legacy offline background map
+  offlineBackgroundMapName: 'Offline Map'
+})
+
+export const useMapStylesQuery = () => {
```

Comment:
can you enforce the typing to `UseQueryResult<MapServerStyleInfo[]>. Both functions are returning that, but since one is not ecplicitly that, 2 different types are coming up in the ts definitions